.github/workflows/dusk_ci.yml

```diff
@@ -19,73 +18,108 @@ jobs:
 
-  build_nightly_nostd:
-    name: Nightly build no_std
+  build_no_std:
+    name: Build no_std
+    strategy:
+      fail-fast: false
+      matrix:
+        toolchain:
+          - nightly
+        target: [ thumbv6m-none-eabi ]
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
+      - name: Check out repository
+        uses: actions/checkout@v2
+
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
         with:
-          profile: minimal  
-          toolchain: nightly
-      # ARM contains no std lib. Just core and alloc.
-      - run: rustup target add thumbv6m-none-eabi
-      - uses: actions-rs/cargo@v1
+          profile: minimal
+          toolchain: ${{ matrix.toolchain }}
+
+      - name: Add target
+        run: rustup target add ${{ matrix.target }}
+
+      - name: Build project with alloc
+        uses: actions-rs/cargo@v1
         with:
           command: build
-          args: --release --no-default-features --features alloc --target thumbv6m-none-eabi
-      - uses: actions-rs/cargo@v1
+          args: --release --no-default-features --features alloc --target ${{ matrix.target }}
+
+      - name: Build project without alloc
+        uses: actions-rs/cargo@v1
         with:
           command: build
-          args: --release --no-default-features --target thumbv6m-none-eabi
+          args: --release --no-default-features --target ${{ matrix.target }}
 
-  test_docs:
-    name: Doc tests
-    runs-on: ubuntu-latest
+  ci:
+    name: Test with all features
+    strategy:
+      fail-fast: false
+      matrix:
+        toolchain:
+          - nightly
+        os:
+          - ubuntu-latest
+          - windows-latest
+    runs-on: ${{ matrix.os }}
     steps:
-      - uses: actions/checkout@v2
-      - run: make doc
-      - run: make doc-internal
+      - name: Check out repository
+        uses: actions/checkout@v2
 
-  fmt:
-      name: Rustfmt
-      runs-on: ubuntu-latest
-      steps:
-        - uses: actions/checkout@v2
-        - uses: actions-rs/toolchain@v1
-          with:
-            profile: minimal
-            toolchain: nightly
-        - run: rustup component add rustfmt
-        - uses: actions-rs/cargo@v1
-          with:
-            command: fmt
-            args: --all -- --check
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ matrix.toolchain }}
+
+      - name: Test project
+        if: ${{ matrix.os != 'ubuntu-latest' || matrix.toolchain != 'nightly' }}
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: --release --all-features
+
+      - name: Test project generating coverage data
+        if: ${{ matrix.os == 'ubuntu-latest' && matrix.toolchain == 'nightly' }}
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: --release --all-features --no-fail-fast
```

Comment:
Do we keep the proper debug symbols in release mode? Just curiosity; I think the coverage tool requires precise debug data in order to evaluate the source mappings

Reply:
I force building with debug symbols in the flags below. Specifically `-Cdebuginfo=2`. That way we keep the test speed benefit of `release` while also being able to profile the code.